ionoscloud/resource_nfs_share_test.go

```diff
@@ -0,0 +1,171 @@
+//go:build all || nfs || nfs_share
+
+package ionoscloud
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services"
+)
+
+func TestAccNFSShare_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		ExternalProviders: map[string]resource.ExternalProvider{
+			"random": {
+				VersionConstraint: "3.4.3",
+				Source:            "hashicorp/random",
+			},
+			"time": {
+				Source:            "hashicorp/time",
+				VersionConstraint: "0.11.1",
+			},
+		},
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckNFSShareDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckNFSShareConfig_basic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckNFSShareExists("ionoscloud_nfs_share.example"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "name", "example-share"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "quota", "512"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "gid", "512"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "uid", "512"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "client_groups.0.description", "Client Group 1"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "client_groups.0.ip_networks.0", "10.234.50.0/24"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "client_groups.0.hosts.0", "10.234.62.123"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "client_groups.0.nfs.0.squash", "all-anonymous"),
+				),
+			},
+			{
+				Config: testAccCheckNFSShareConfig_update,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckNFSShareExists("ionoscloud_nfs_share.example"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "name", "example-share-updated"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "quota", "1024"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "gid", "1024"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "uid", "1024"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "client_groups.0.description", "Client Group 1 Updated"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "client_groups.0.ip_networks.0", "10.234.50.0/24"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "client_groups.0.hosts.0", "10.234.62.124"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "client_groups.0.nfs.0.squash", "root-anonymous"),
+				),
+			},
+			{
+				Config: testAccDataSourceNFSShareMatchId,
```

Comment:
@cristiGuranIonos maybe also add a test for name match and partial match.

Reply:
more complex tests TBD after resources are available